main.tf

```diff
@@ -120,17 +120,17 @@ resource "aws_route_table" "quortex_private" {
 
   # Route to the NAT, if NAT is enabled...
   dynamic "route" {
-    for_each = local.enable_nat_gateway ? [1] : []
+    for_each = local.availability_zones["pub-${each.value.availability_zone}"] != null ? [1] : []
```

Comment:
Why the key is prefixed by "pub-" ?

Reply:
Good catch, this is due to the nat_gateways variable passed to this module from the other repo being a map of public subnets (with the pub prefix in its keys), but there is no warranty that this will be the case in other usage.
I'll rework the map.